packages/ui/src/redux/slices/settings-dialog.ts

```diff
@@ -25,6 +27,9 @@ const settingsDialogSlice = createSlice({
     setOnChangingPassword: (state: Draft<SettingsDialog>, action: PayloadAction<boolean>) => {
       state.onChangingPassword = action.payload;
     },
+    setOpen: (state: Draft<SettingsDialog>, action: PayloadAction<boolean>) => {
+      state.open = action.payload;
+    },
```

Comment:
If we introduce a new a redux `open` state here, I think we can handle it such that we can remove the `setTimeout(() => dispatch(settingsDialogActions.resetState()), 150);` above. Something like below:
```suggestion
    open: (state: Draft<SettingsDialog>) => {
      // reset state before open modal
      state.screen = SettingsDialogScreen.SettingsWallet;
      state.onChangingPassword = false;
      state.verifiedPassword = null;
      state.open = true;
    },
    hide: (state: Draft<SettingsDialog>) => {
       state.open = false;
    }
```
Can you try it out to see if it works?